Formula/kong.rb

```diff
@@ -40,17 +40,42 @@ def install
   end
 end
 
-# patch Kong default `prefix` to `/usr/local/opt/kong` as `/usr/local/`
-# not writable by non root user on OSX
+# patch Kong default `prefix` to `HOMEBREW_PREFIX/opt/kong`
+# to ensure it's writeable
+# additionally, add brew on m1 paths to lua_path and lua_cpath
 __END__
+diff --git a/bin/kong b/bin/kong
+index 3e0ecf97d..b03e18a23 100755
+--- a/bin/kong
++++ b/bin/kong
+@@ -4,6 +4,7 @@ setmetatable(_G, nil)
+ 
+ pcall(require, "luarocks.loader")
+ 
+-package.path = (os.getenv("KONG_LUA_PATH_OVERRIDE") or "") .. "./?.lua;./?/init.lua;" .. package.path
++package.cpath = (os.getenv("KONG_LUA_CPATH_OVERRIDE") or "") .. "/opt/homebrew/lib/lua/5.1/?.so;" .. package.cpath
++package.path = (os.getenv("KONG_LUA_PATH_OVERRIDE") or "") .. "./?.lua;./?/init.lua;" .. "/opt/homebrew/share/lua/5.1/?.lua;/opt/homebrew/share/lua/5.1/?/init.lua;" .. package.path
+ 
+ require("kong.cmd.init")(arg)
 diff --git a/kong/templates/kong_defaults.lua b/kong/templates/kong_defaults.lua
-index e38b475..7a74a2f 100644
+index 5937dad10..c3387fded 100644
 --- a/kong/templates/kong_defaults.lua
 +++ b/kong/templates/kong_defaults.lua
 @@ -1,5 +1,5 @@
  return [[
 -prefix = /usr/local/kong/
-+prefix = /usr/local/opt/kong/
++prefix = HOMEBREW_PREFIX/opt/kong/
  log_level = notice
  proxy_access_log = logs/access.log
  proxy_error_log = logs/error.log
+@@ -166,8 +166,8 @@ lua_socket_pool_size = 30
+ lua_ssl_trusted_certificate = NONE
+ lua_ssl_verify_depth = 1
+ lua_ssl_protocols = TLSv1.1 TLSv1.2 TLSv1.3
+-lua_package_path = ./?.lua;./?/init.lua;
+-lua_package_cpath = NONE
++lua_package_path = ./?.lua;./?/init.lua;/opt/homebrew/share/lua/5.1/?.lua;/opt/homebrew/share/lua/5.1/?/init.lua
++lua_package_cpath = /opt/homebrew/lib/lua/5.1/?.so;
```

Comment:
let's use `;;` at the end, so we are adding new search path but not deleting old ones. same aplies to lua_package_path